sdk/identity/azure-identity/README.md

```diff
@@ -81,6 +81,20 @@ The [Managed identity authentication](https://docs.microsoft.com/azure/active-di
 * [Azure Cloud Shell](https://docs.microsoft.com/azure/cloud-shell/msi-authorization)
 * [Azure Arc](https://docs.microsoft.com/azure/azure-arc/servers/managed-identity-authentication)
 
+## Multi tenant authentication support 
+
+Added support for multi tenant authentication. 
+If the client is enabled for multi tenant authentication to use the feature you will need to pass a Azure::Identity::ChallengeClientSecretCredential at client creation instead of ClientSecretCredential, otherwise the single tenant authentication is used. 
```

Comment:
yes i know, the idea was that i didn't want to change the existing gettoken that is used in bearertoken implementation. this way we can transition one sdk at a time to use multi tenant auth instead of changing the existing auth behavior.

Reply:
I see no reason to deviate from other SDKs. Plus, if you make existing credentials to support challenge-based auth, then users can use EnvironmentCredential, for example. Or the Managed Identity Credential. With this design, if a client is using Clallenge-Based auth, it won't work with Managed Identity cred.